src/components/CentralHeader/CentralHeader.tsx

```diff
@@ -31,22 +32,42 @@ interface ICentralHeaderProps extends RouteComponentProps {
 /**
  * Central header right below Slim header, containing app title and user name and user role with dropdown to go to profile page and logout
  */
-export const CentralHeader = withRouter<
-  ICentralHeaderProps,
-  React.FC<ICentralHeaderProps>
->((props: ICentralHeaderProps) => {
+export const CentralHeader = withRouter((props: ICentralHeaderProps) => {
   /**
    * react-i18next translation hook
    */
   const { t } = useTranslation();
 
+  /**
+   * add user icon if user name is set
+   */
+  const userIcon =
```

Comment:
consider also to make a `<UserIcon >` component that takes the username as input prop

Reply:
I see the advantage of creating child components for a more ordered code and to give each component it's logic, but in this case the amount of code is quite small...which would be the advantage of creating a component for such few lines of code?

Reply:
I see the same advantage of breaking code into smaller pieces: it favors testability (think about: how would you test these lines here ?) composability and reusability (what if I have to put an userIcon somewhere else and then change appearance ?). Now you have a big component (CentralHeader) that spans 150 lines of code. I'm not saying you have to create another file, just put these lines outside `CentralHeader` in the same source; you'll end up with a shorter method and your code will be more declarative. While this suggestion may be negligibile in this specific context, it is a general advice : -)